package cms_8120;
option java_package = "com.hikvision.finance.fms.cms.hpp.csclient";
option java_outer_classname = "ProductTypeProto";
option java_generic_services = true;

message ProductType
{
    //必须字段
    optional int32 id                       = 1;    // 在数据库中的流水号
    optional int32 pid                      = 2;    // 父类型ID
    optional string strName                 = 3;    // 类型名称
    optional string strInterAreaCode        = 4;    // 区域编码
    optional string strPath                 = 5;    // 节点路径
    optional int32 nsortId                  = 6;    // 同级排序号
}

message ReqProductType
{
    optional int32 cmd                	    = 1;    // CMD_DT_PRODUCTTYPE_LIST
    optional int32 id                       = 2;    // 在数据库中的流水号
    optional int32 pid                      = 3;    // 父类型ID
    optional string strName                 = 4;    // 类型名称
    optional string strInterAreaCode        = 5;    // 区域编码
    optional string strPath                 = 6;    // 节点路径
    optional int32 nsortId                  = 7;    // 同级排序号 
}

message ProductTypeList
{
    optional int32 cmd                      = 1;    // CMD_DT_PRODUCTTYPE_LIST
    optional int32 result                   = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info             = 3;    // 信息
    repeated ProductType productType        = 4;    // 产品类型列表
}

service ProductTypeService {
	
	rpc GetProductTypeList(ReqProductType) returns (ProductTypeList);	// CMD_DT_PRODUCTTYPE_LIST(获取所有产品类型信息)
	
}

